hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -200,6 +199,22 @@ public RpcClient(ConfigurationSource conf, String omServiceId)
       // requests serviced by this client will need S3 Auth set.
       ozoneManagerProtocolClientSideTranslatorPB.setS3AuthCheck(
           conf.getBoolean(S3Auth.S3_AUTH_CHECK, false));
+      String omVersion = conf.get(OZONE_OM_CLIENT_PROTOCOL_VERSION_KEY);
```

Comment:
We need the actual version string, going forward we can have higher versions so the presence is not enough.
Also, it is made configurable in case we need to override the version string in a deployment for any reason.

Reply:
We need the actual version string, going forward we can have higher versions so the presence is not enough.

For now we should be good. And as said old client talking to new server and new client talking to old server should be supported. So, technically clients based on version returned by server and  not talking i think we donot need it.

Reply:
I have moved to a version comparison scheme and should protect in a future safe way for only secure S3G communication.